New Resource: vault_gcp_auth_backend_role

angrylogic opened the pull request:
This PR enables managment of Vault GCP Auth backend roles.

petems commented:
@angrylogic I've been using this fork for a bit, works well! 😄 

Have you thought about extending it to also configure the GCP auth config as well? So actually giving it the credentials file: https://www.vaultproject.io/api/auth/gcp/index.html#configure

So doing this step but with Terraform:

```
vault auth enable 'gcp'
vault write auth/gcp/config credentials=@vault_auth_checker_credentials.json
```

Then, you could do the entire process of Vault in GCP with Terraform:

```hcl
resource "vault_auth_backend" "gcp" {
    path = "gcp"
    type = "gcp"
}

resource "google_service_account_key" "vault_auth_checker_credentials" {
  service_account_id = "${google_service_account.vault_auth_checker.name}"
  public_key_type = "TYPE_X509_PEM_FILE"
}

resource "vault_auth_backend_gcp_config" gcp {
   credentials = "${base64decode(google_service_account_key.vault_auth_checker_credentials.private_key)}"
}
```

fatmcgav commented:
We're interested in this aswell... Any plans to get this merged?

tyrannosaurus-becks commented on vault/resource_gcp_auth_backend_role.go:
```diff
@@ -0,0 +1,218 @@
+
+	log.Printf("[DEBUG] Writing role %q to GCP auth backend", path)
+	_, err := client.Logical().Write(path, data)
+	d.SetId(path)
```
Would it be possible to move this up a line so it's not between the err and the err check?

angrylogic commented:
@tyrannosaurus-becks  -- rebased and addressed feedback in this one as well.

Tests passing for me:
```
$ go test -test.v -run TestGCPAuthBackendRole_basic
=== RUN   TestGCPAuthBackendRole_basic
--- PASS: TestGCPAuthBackendRole_basic (0.18s)
PASS
ok      github.com/terraform-providers/terraform-provider-vault/vault   0.300s
```

Thanks!
